geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveAttributeStore.scala

```diff
@@ -0,0 +1,256 @@
+import org.apache.hadoop.mapreduce.Job
+import org.apache.log4j.Logger
+import org.apache.spark.{Logging, SparkConf, SparkContext}
+import org.apache.spark.SparkContext._
+import org.geotools.coverage.grid._
+import org.geotools.gce.geotiff._
+import org.opengis.coverage.grid.GridCoverage
+import org.opengis.parameter.GeneralParameterValue
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+import resource._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+
+object GeowaveAttributeStore {
+
+  def accumuloRequiredOptions(
+    zookeepers: String,
+    accumuloInstance: String,
+    accumuloUser: String,
+    accumuloPass: String,
+    geowaveNamespace: String
+  ): AccumuloRequiredOptions = {
+    val aro = new AccumuloRequiredOptions
+    aro.setZookeeper(zookeepers)
+    aro.setInstance(accumuloInstance)
+    aro.setUser(accumuloUser)
+    aro.setPassword(accumuloPass)
+    aro.setGeowaveNamespace(geowaveNamespace)
+    aro
+  }
+
+  def basicOperations(
+    zookeepers: String,
+    accumuloInstance: String,
+    accumuloUser: String,
+    accumuloPass: String,
+    geowaveNamespace: String
+  ): BasicAccumuloOperations = {
+    return new BasicAccumuloOperations(
+      zookeepers,
+      accumuloInstance,
+      accumuloUser,
+      accumuloPass,
+      geowaveNamespace)
+  }
+
+  def adapters(bao: BasicAccumuloOperations): Array[RasterDataAdapter] = {
+    for (adapters <- managed(new AccumuloAdapterStore(bao).getAdapters)) {
+      val list = Iterator.iterate(adapters)({ _ => adapters })
+        .takeWhile({ _ => adapters.hasNext })
+        .map(_.next.asInstanceOf[RasterDataAdapter])
+        .toArray
+
+      return list
+    }
+    Array.empty[RasterDataAdapter]
+  }
+
+  def primaryIndex = (new SpatialDimensionalityTypeProvider.SpatialIndexBuilder).createIndex()
+
+  def subStrategies(idx: PrimaryIndex) = idx
+    .getIndexStrategy
+    .asInstanceOf[HierarchicalNumericIndexStrategy]
+    .getSubStrategies
+
+}
+
+class GeowaveAttributeStore(
+  zookeepers: String,
+  accumuloInstance: String,
+  accumuloUser: String,
+  accumuloPass: String,
+  geowaveNamespace: String
+) extends DiscreteLayerAttributeStore with Logging {
+
+  def zookeepers(): String = zookeepers
+  def accumuloInstance(): String = accumuloInstance
+  def accumuloUser(): String = accumuloUser
+  def accumuloPass(): String = accumuloPass
+  def geowaveNamespace(): String = geowaveNamespace
+
+  val zkInstance = (new ZooKeeperInstance(accumuloInstance, zookeepers))
+  val token = new PasswordToken(accumuloPass)
+  val connector = zkInstance.getConnector(accumuloUser, token)
+  val delegate = AccumuloAttributeStore(connector, s"${geowaveNamespace}_ATTR")
+
+  val bao = GeowaveAttributeStore.basicOperations(
+    zookeepers: String,
+    accumuloInstance: String,
+    accumuloUser: String,
+    accumuloPass: String,
+    geowaveNamespace: String
+  )
+  val aro = GeowaveAttributeStore.accumuloRequiredOptions(
```

Comment:
More verbose names would fit with the library patterns better, and IMO make the code much more readable.